core/src/main/java/org/apache/brooklyn/core/config/BasicConfigKey.java

```diff
@@ -331,12 +337,19 @@ public ConfigInheritance getParentInheritance() {
         return parentInheritance;
     }
 
+    /** @see ConfigKey#getConstraints() */
+    @Nonnull
+    @Override
+    public Map<String, Predicate<? super T>> getConstraints() {
+        return constraints;
+    }
+
     /** @see ConfigKey#getConstraint() */
     @Override @Nonnull
     public Predicate<? super T> getConstraint() {
         // Could be null after rebinding
-        if (constraint != null) {
-            return constraint;
+        if (constraints.size() > 0) {
+            return Predicates.and(constraints.values());
         } else {
             return Predicates.alwaysTrue();
         }
```

Comment:
Thought about that but there is issue as you can have more than one implementation of `Predicate` to do the same thing. In the case of `required`, I have seen implemented as:
- `StringPredicates.isNonBlank()`
- `Predicates.notNull()`
And we could potential have others